app/code/Ometria/Core/Plugin/OrderManagementPlugin.php

```diff
@@ -0,0 +1,112 @@
+<?php
+namespace Ometria\Core\Plugin;
+
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Api\OrderManagementInterface;
+use Ometria\Core\Helper\Config;
+use Ometria\Core\Model\Config\Source\StockPushScope;
+use Ometria\Core\Service\Product\Inventory as InventoryService;
+use Ometria\Core\Service\PushApi as PushApiService;
+
+class OrderManagementPlugin
+{
+    /** @var InventoryService */
+    private $inventoryService;
+
+    /** @var PushApiService */
+    private $pushApiService;
+
+    /** @var Config */
+    private $helperConfig;
+
+    /**
+     * @param InventoryService $inventoryService
+     * @param PushApiService $pushApiService
+     * @param Config $helperConfig
+     */
+    public function __construct(
+        InventoryService $inventoryService,
+        PushApiService $pushApiService,
+        Config $helperConfig
+    ) {
+        $this->inventoryService = $inventoryService;
+        $this->pushApiService = $pushApiService;
+        $this->helperConfig = $helperConfig;
+    }
+
+    /**
+     * Plugin to trigger webhook to Ometria where stock value reaches 0 for
+     * an MSI sales channel after an order is placed.
+     *
+     * @param OrderManagementInterface $subject
+     * @param $result
+     * @param OrderInterface $order
+     * @return mixed
+     */
+    public function afterPlace(OrderManagementInterface $subject, $result, OrderInterface $order)
+    {
+        try {
+            $this->sendPushNotifications($order);
+        } catch (\Exception $e) {
+            // Catch all errors to ensure this does not affect order placement
```

Comment:
shouldn't we at least error log the exception ?
I know that those logs will not go to us

but it may help clients to identify some issues on their side

Reply:
@mkuzdowicz In previous discussions around logging from the Ometria extension there has been a preference not to write logs to mitigate the possibility of bringing sites down by filling the disk. We have previously added an Ometria specific log file for this type of thing however it was removed after discussion of risks. I agree that logging in these scenarios is absolutely the right thing to do and would make the often difficult task of debugging the client's issues a lot simpler.